### ISSUE TITLE
Wrong response on API create conversation

### ISSUE THREAD

**Issue by fulldecent** [freescout-helpdesk/freescout#2495]
The documentation specifies a HTTP 201 response code with an empty response for the create conversation method on success.

![Screenshot 2022-12-30 at 10 53 33](https://user-images.githubusercontent.com/382183/210088839-c3532894-b834-488b-abf4-d16e514a05f8.png)

However in the application it is sending a 200 response code with some whitespace output.

**Comment by fulldecent** [comment]
I'm getting this response with HTTP/200:

```
Server: nginx/1.21.4
Content-Type: application/json
Connection: keep-alive
Cache-Control: max-age=0, must-revalidate, no-cache, no-store, private
Date: Fri, 30 Dec 2022 16:00:38 GMT
Pragma: no-cache
```

**Comment by freescout-helpdesk** [comment]
Here is how it looks:
![2022-12-31_08-59-54](https://user-images.githubusercontent.com/40499291/210126836-e8333313-5938-4d06-b7a7-27a1b70d2551.png)

Make sure you are using the latest version of the API & Webhooks Module.

**Comment by fulldecent** [comment]
Have the latest versions.

Maybe there is some error condition that produces empty content like ` ` but not a 201. There's a few ways to return a response in that endpoint, I'm looking through the code and can't see which actual response is being applied.

**Comment by freescout-helpdesk** [comment]
Is conversation created when you get HTTP/200 response?

**Comment by freescout-helpdesk** [comment]
Can you try to reproduce the issue on https://demo.freescout.net/api/ and send us request body.

**Comment by fulldecent** [comment]
Conversation is not created. Have not tried against demo site yet. But here is my request:

```php
    $requestObject = [
        'type' => 'phone',
        'mailboxId' => '2',
        'subject' => 'SMS from ' . $did . ' to ' . $dst . ' ' . $recipient . ' (' . $month . ')',
        'customer' => [
            'firstName' => $recipient,
            'phone' => $dst,
        ],
        'threads' => [
            [
                'type' => 'note',
                'text' => 
                    'This is a SMS conversation between ' . $did . ' and ' . $dst . "\n\n"
                'user' => 14, // Posted by the robot account
            ],
        ],
        'imported' => true, // Suppress email notifications
        //'assignTo' => 18, // Always assign to Honey
        'assignTo' => 1, // Always assign to Will
        'status' => 'pending',
    ];
```

**Comment by freescout-helpdesk** [comment]
Is there anything in Manage > Logs > App Logs?

**Comment by freescout-helpdesk** [comment]
We can't reproduce the issue. Try to disable all non-official modules and debug `/Modules/ApiWebhooks/Http/Controllers/ApiController.php` (line 1313)

**Comment by fulldecent** [comment]
Here is a reproduction use case demonstrating problem against the test server:

```php
<?php

// PERFORM REQUEST /////////////////////////////////////////////////////////////
$target = 'https://demo.freescout.net/api/conversations';
$apiKey = # get this from https://demo.freescout.net/app-settings/apiwebhooks

$ch = curl_init();
$requestObject = [
    'type' => 'phone',
    'mailboxId' => '2',
    'subject' => 'a subj',
    'customer' => [
        'firstName' => 'aFirstName',
        'phone' => '0123456789',
    ],
    'threads' => [
        [
            'type' => 'note',
            'text' => 'a text note',
            'user' => 1, // some user number that exists
        ],
    ],
    'imported' => true, // Suppress email notifications
    'assignTo' => 1, // assign to some user that exists
    'status' => 'pending',
];
curl_setopt_array($ch, [
    CURLOPT_URL => $target,
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_POST => true,
    CURLOPT_HEADER => true,
    CURLOPT_NOBODY => true,
    CURLOPT_POSTFIELDS => json_encode($requestObject),
    CURLOPT_HTTPHEADER => [
        'Content-Type: application/json',
        'X-FreeScout-API-Key: ' . $apiKey,
    ],
]);

// REPORT ON OUTPUT ////////////////////////////////////////////////////////////
$result = curl_exec($ch);
$httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
$resultHeader = substr($result, 0, curl_getinfo($ch, CURLINFO_HEADER_SIZE));
$resultBody = substr($result, curl_getinfo($ch, CURLINFO_HEADER_SIZE));

header('Content-Type: text/plain');
echo "# HTTP Code" . PHP_EOL;
echo $httpCode . PHP_EOL;
echo "# Result Header" . PHP_EOL;
echo $resultHeader . PHP_EOL;
echo "# Result Body" . PHP_EOL;
echo $resultBody . PHP_EOL;

curl_close($ch);
```

Result of running `php make-convo.php`:

```
# HTTP Code
200
# Result Header
HTTP/1.1 200 OK
Server: nginx/1.14.0 (Ubuntu)
Content-Type: application/json
Connection: keep-alive
Cache-Control: max-age=0, must-revalidate, no-cache, no-store, private
Date: Thu, 05 Jan 2023 21:56:15 GMT
Pragma: no-cache


# Result Body

```

**Comment by freescout-helpdesk** [comment]
API returns 201 code and the body in response to this request. To get the body comment the following line:
```
//CURLOPT_NOBODY => true,
```

It should look like:
```
...
curl_setopt_array($ch, [
    CURLOPT_URL => $target,
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_POST => true,
    CURLOPT_HEADER => true,
    //CURLOPT_NOBODY => true,
    CURLOPT_POSTFIELDS => json_encode($requestObject),
    CURLOPT_HTTPHEADER => [
        'Content-Type: application/json',
        'X-FreeScout-API-Key: ' . $apiKey,
    ],
]);

// REPORT ON OUTPUT ////////////////////////////////////////////////////////////
$result = curl_exec($ch);
$httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
$resultHeader = substr($result, 0, curl_getinfo($ch, CURLINFO_HEADER_SIZE));
//$resultBody = substr($result, curl_getinfo($ch, CURLINFO_HEADER_SIZE));

header('Content-Type: text/plain');
echo "# HTTP Code" . PHP_EOL;
echo $httpCode . PHP_EOL;
echo "# Result Header" . PHP_EOL;
echo $resultHeader . PHP_EOL;
echo "# Result Body" . PHP_EOL;
echo $result . PHP_EOL;

curl_close($ch);
```

**Comment by fulldecent** [comment]
Well fuck. Today I learned that `CURLOPT_POST` and `CURLOPT_NOBODY` are conflicting. The former sets method to `POST` and the latter sets it to `HEAD`. Thank you.

Okay so actually I was calling:

> HEAD /api/conversations

which is not an API endpoint. So here is the actual minimal endpoint.

```php
<?php
// PERFORM REQUEST /////////////////////////////////////////////////////////////
$target = 'https://demo.freescout.net/api/conversations';
$apiKey = # get this from https://demo.freescout.net/app-settings/apiwebhooks

$ch = curl_init();
curl_setopt_array($ch, [
    CURLOPT_URL => $target,
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_POST => true,
    CURLOPT_HEADER => true,
    CURLOPT_NOBODY => true,
    CURLINFO_HEADER_OUT => true,
    CURLOPT_HTTPHEADER => [
        'Content-Type: application/json',
        'X-FreeScout-API-Key: ' . $apiKey,
    ],
]);

// REPORT ON OUTPUT ////////////////////////////////////////////////////////////
$result = curl_exec($ch);
$info = curl_getinfo($ch);
curl_close($ch);

$responseHeaders = substr($result, 0, $info['header_size']);
$responseBody = substr($result, $info['header_size']);

header('Content-Type: text/plain');
echo '# Request headers' . PHP_EOL;
echo $info['request_header'] . PHP_EOL . PHP_EOL;
echo '# Response headers' . PHP_EOL;
echo $responseHeaders . PHP_EOL . PHP_EOL;
echo '# Response body' . PHP_EOL;
echo $responseBody . PHP_EOL . PHP_EOL;
```

---

I guess the server should correctly be returning a 4xx response here because `HEAD /api/conversations` is not an endpoint.

**Comment by freescout-helpdesk** [comment]
Fixed in API & Webhooks Module v1.0.47
